packages/react-router/lib/dom/ssr/browser.tsx

```diff
@@ -105,6 +114,33 @@ function createHydratedRouter(): RemixRouter {
     ssrInfo.context.isSpaMode
   );
 
+  // If the user provided their own data routes, add them to our manifest-created
+  // routes here before creating the data router
+  let propRoutesError: unknown;
+  if (propRoutes && propRoutes.length > 0) {
+    let rootRoute = routes[0];
+    if (!rootRoute.children) {
+      rootRoute.children = [];
+    }
+
+    try {
+      validatePropRoutes(propRoutes, rootRoute.children);
+
+      // If a route doesn't have a loader, add a dummy hydrating loader to stop
+      // rendering at that level for hydration
+      let hydratingLoader: LoaderFunction = () => null;
+      hydratingLoader.hydrate = true;
+      for (let route of propRoutes) {
+        if (!route.loader) {
+          route = { ...route, loader: hydratingLoader };
+        }
+        rootRoute.children.push(route);
```

Comment:
TODO: Work through some scenarios here - do we _need_ a dummy hydrating loader to prevent hydration issues?  Also do we need to add `.hydrate` to the user-provided loader?